AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Subscribe/DataStoreObserveQueryOperationTests.swift

```diff
@@ -77,6 +77,125 @@ class DataStoreObserveQueryOperationTests: XCTestCase {
     }
 
+    ///  ObserveQuery will send a single snapshot when the sync state toggles
+    ///  from false to true. The operation internally listens to `.modelSynced` event from
+    ///  the Hub.
+    ///
+    /// - Given: ObserveQuery has started and the first snapshot has been received.
+    /// - When:
+    ///    - modelSyncedEvent is sent to the Hub
+    /// - Then:
+    ///    - ObserveQuery will send a second snapshot
+    ///
+    func testGenerateSnapshotOnObserveQueryWhenModelSynced() throws {
+        let firstSnapshot = expectation(description: "first query snapshots")
+        let secondSnapshot = expectation(description: "second query snapshots")
+        let thirdSnapshot = expectation(description: "third query snapshot")
+        thirdSnapshot.isInverted = true
+        
+        var querySnapshots = [DataStoreQuerySnapshot<Post>]()
+        let dispatchedModelSyncedEvent = AtomicValue(initialValue: false)
+        let operation = AWSDataStoreObserveQueryOperation(
+            modelType: Post.self,
+            modelSchema: Post.schema,
+            predicate: nil,
+            sortInput: nil,
+            storageEngine: storageEngine,
+            dataStorePublisher: dataStorePublisher,
+            dataStoreConfiguration: .default,
+            dispatchedModelSyncedEvent: dispatchedModelSyncedEvent)
+
+        let sink = operation.publisher.sink { completed in
+            switch completed {
+            case .finished:
+                break
+            case .failure(let error):
+                XCTFail("Failed with error \(error)")
+            }
+        } receiveValue: { querySnapshot in
+            querySnapshots.append(querySnapshot)
+            if querySnapshots.count == 1 {
+                firstSnapshot.fulfill()
+            } else if querySnapshots.count == 2 {
+                secondSnapshot.fulfill()
+            } else if querySnapshots.count == 3 {
+                XCTFail("Should not receive third snapshot for a Model change")
+                thirdSnapshot.fulfill()
+            }
+        }
+        let queue = OperationQueue()
+        queue.addOperation(operation)
+        wait(for: [firstSnapshot], timeout: 5)
+
+        dispatchedModelSyncedEvent.set(true)
+        let modelSyncedEventPayload = HubPayload(eventName: HubPayload.EventName.DataStore.modelSynced,
+                                                 data: ModelSyncedEvent(modelName: Post.modelName,
+                                                                        isFullSync: true, isDeltaSync: false,
+                                                                        added: 0, updated: 0, deleted: 0))
+        Amplify.Hub.dispatch(to: .dataStore, payload: modelSyncedEventPayload)
+        wait(for: [secondSnapshot], timeout: 10)
+
+        let modelSyncedEventNotMatch = HubPayload(eventName: HubPayload.EventName.DataStore.modelSynced,
+                                                  data: ModelSyncedEvent.Builder().modelName)
```

Comment:
The intention of passing in a "not matching" model synced event should at least have the correct data payload of a ModelSyncedEvent, instead of just a string payload. Right now `ModelSyncedEvent.Builder().modelName` is of type String, with an empty string value (since the Builder() method initializes the ModelSyncedEvent.modelName with an empty string). What you'll want here is to create an instance of ModelSyncedEvent.Builder, update the modelName, and then call `build()` to create an instance of ModelSyncedEvent, and pass that to the HubPayload